src/services/events/utils/types.ts

```diff
@@ -0,0 +1,119 @@
+import { CartDto, Product, User } from '../../../interfaces'
+
+export type EventCallbackFunction<Payload = undefined> = (payload: Payload) => void
+
+export type GetCallbackPayload<T> = T extends EventCallbackFunction<infer Payload>
+  ? Payload
+  : undefined
+
+export enum HeseyaEventType {
+  addToCart = 'addToCart',
```

Comment:
Wartości enuma z dużej litery
```suggestion
  AddToCart = 'addToCart',
```